### ISSUE TITLE
ESLint cause error with correct Regexp input

### ISSUE THREAD

**Issue by juggernautjp** [eslint/eslint#14747]
**Tell us about your environment**

- Node version: v14.17.1
- npm version: v6.14.13
- Local ESLint version: v7.27.0
- Global ESLint version: v7.29.0 (Currently used)
- Operating System: win32 10.0.19043

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default (I think)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

.eslintrc.yml:

```js
---
# _source: http://amobiz.github.io/2015/12/21/applying-eslint-using-atom-linter-eslint/
# _log:
#   - '2017-03-02: first edit'
#   - '2017-07-12: port from json to yaml; port to unify my favorite style along with work env'
#   - '2017-07-18': update with es6 style
#   - '2021-04-17': integrate with prettier v2
# _NOTE: copy this file to folder you want; rename it as .eslintrc.*; if possible reload text editor;
root: true
env:
  browser: true
  commonjs: true
  es6: true
  node: true
globals:
  atom: true
  describe: true
  it: true
extends:
  - 'eslint:recommended'
  - 'prettier'
parserOptions:
  ecmaVersion: 8
  sourceType: module
rules:
  no-console: off
  no-unused-vars:
    - warn
    - vars: all
      args: after-used
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

validation.js

```js
121:                  const specialChr = /(?<!`|`!)[~@#%^*]/g;
122:                  const _str = flattenCson[k].toString();
123:                  const _res = _str.search(specialChr);
124:                  const errMsg = `\n\tfound special chr: '${_str[_res]}' in value: '${_str}'\n\n\tcson-path: '${k}'\t`;
```

```bash
$ eslint spec/validation.js
```

**What did you expect to happen?**

The above code dose work correctly, as the following cases:
1. match the string "*.js"
2. dose not match the string "\`*.js\`"
3. dose not match the string "\`!*.json\`"

**What actually happened? Please include the actual, raw output from ESLint.**

With the above code, ESLint cause the following Error:
```js
  121:39  error  Parsing error: Invalid regular expression: /(?<!`|`!)[~@#%^*]/: Invalid group

✖ 1 problem (1 error, 0 warnings)
```

**Comment by mdjermanovic** [comment]
Hi @juggernautjp, thanks for the issue!

Lookbehind assertions are ES2018, can you update your config file to set `ecmaVersion: 9` instead of `ecmaVersion: 8` (which is equivalent to 2017).

**Comment by juggernautjp** [comment]
Thank you very much. It dose work with no error, by replacing `ecmaVersion: 8` with `ecmaVersion: 9`.
Your advice fixed this bug and this issue is closed.
